src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -33,13 +35,23 @@ import scala.reflect.ClassTag
 import scala.util.control.Breaks._
 import com.microsoft.azure.cosmosdb.rx.internal.NotFoundException
 
+private object CosmosDBConnection {
+  private val rnd = scala.util.Random
+
+  def getRandomRetryInterval(retryAttempt: Int): Int = {
+    rnd.nextInt(math.min(retryAttempt * 100, 5000))
```

Comment:
why do we need a random retry?

Reply:
Discussed offline - in the customer case ReadTimeout happened when getting PkRanges - adding random timeout to reduce chance that retries are happening at the same time